app/helpers/results_helper.rb

```diff
@@ -13,21 +13,29 @@ def result_groups(session)
   end
 
   def filter_questions_by_session(group_key, session)
-    I18n.t("results_link.#{group_key}").each_with_object([]) do |question, array|
+    group = I18n.t("results_link.#{group_key}").dup
+    group.each_with_object([]) do |question, array|
       if question[1][:show_options].include?(session[question[0]])
-        array << I18n.t("results_link.#{group_key}.#{question[0]}")
+        array << filter_results_by_country(I18n.t("results_link.#{group_key}.#{question[0]}").dup)
       end
     end
   end
 
+  def filter_results_by_country(question_results)
```

Comment:
A little on the nitpick side, but I think `nation` is a better word to use than `country` (and seems to be the word we use on GOV.UK)

Reply:
done, removed it from method names and commit messages